spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java

```diff
@@ -61,4 +65,42 @@ public static void validatePartitionTransforms(PartitionSpec spec) {
           String.format("Cannot write using unsupported transforms: %s", unsupported));
     }
   }
+
+  /**
+   * A modified version of Spark's LookupCatalog.CatalogAndIdentifier.unapply
+   * Attempts to find the catalog and identifier a multipart identifier represents
+   * @param nameParts Multipart identifier representing a table
+   * @return The CatalogPlugin and Identifier for the table
+   */
+  public static <C, T> Pair<C, T> catalogAndIdentifier(List<String> nameParts,
+                                                       Function<String, C> catalogProvider,
+                                                       IdentiferFunction<T> identiferProvider,
+                                                       C defaultCatalog,
```

Comment:
Current catalog? I think it would be current catalog and current namespace.

Reply:
It is the default catalog in `Spark3Util` on master. Is that wrong? current catalog seems more appropriate to me...

Reply:
I think it's okay. The logic is correct.